### PR TITLE
Logging

### PR DESCRIPTION
**Pull request by glinares-em**
#310

### CODE REVIEW

File: examples/bokeh_handler.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import logging
```
Comment (redhog) [inline examples/bokeh_handler.py]: This file should not be included, as this functionality exists in bokeh_garden/logging_handler.py above

File: examples/grib_info_download.py [hunk 2]
```diff
@@ -5,11 +5,30 @@
...
 import bokeh.models.sources
 import io
+import logging
+
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+logger.setLevel(logging.ERROR)
+
+# logger = logging.getLogger(__name__)
```
Comment (redhog) [inline examples/grib_info_download.py]: Please remove the commented out code as well as the _file_handler that you're not really using...

File: examples/grib_log.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
+import bokeh.io
```
Comment (redhog) [inline examples/grib_log.py]: This file should not be included, as this functionality exists in bokeh_garden/logging_bg.py above

File: examples/grib_plot.py [hunk 4]
```diff
@@ -3,26 +3,59 @@
 import pandas as pd
 import tempfile
 import pygrib
+import logging
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+#formatter = logging.Formatter('%(asctime)s:%(levelname)s:%(name)s:%(message)s')
```
Comment (redhog) [inline examples/grib_plot.py]: Please remove commented out code here

File: examples/grib_plot.py [hunk 5]
```diff
@@ -3,26 +3,59 @@
...
         self._source = bokeh.models.ColumnDataSource(data={"images": [], "x": [], "y": [], "dw": [], "dh":[]})
+        print("D source", self._source)
         self._figure = bokeh.plotting.Figure()
+        print("E figure", self._figure)
         self._figure.image("images", x="x", y="y", dw="dw", dh="dh", palette="Spectral11", source=self._source)
 
+
         bokeh.models.layouts.Column.__init__(self, self._sel, self._figure, **kw)
-        
-        self._app.on_change("grib", self.grib_changed)
```
Comment (redhog) [inline examples/grib_plot.py:18]: Please change this line back to how it used to be...

File: examples/grib_upload.py [hunk 6]
```diff
@@ -4,22 +4,66 @@
...
 import bokeh.models.sources
+import logging
+import examples.bokeh_handler
+#from bokeh.io import show
+#from bokeh.models import Div
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+logger.setLevel(logging.ERROR)
+# logger.setLevel(logging.DEBUG)
```
Comment (redhog) [inline examples/grib_upload.py]: Please remove commented out lines

File: examples/weather_data_explorer.py [hunk 7]
```diff
@@ -3,16 +3,32 @@
...
 import examples.grib_upload
 import examples.grib_info_download
+import logging
+import examples.grib_log
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+formatter = logging.Formatter('%(asctime)s:%(levelname)s:%(name)s:%(message)s')
+
+#file_handler = logging.FileHandler('test2.log')
```
Comment (redhog) [inline examples/weather_data_explorer.py]: Same here, please remove commented out stuff

File: examples/weather_data_explorer.py [hunk 8]
```diff
@@ -3,16 +3,32 @@
...
+# file_handler.setFormatter(formatter)
+#logger.addHandler(file_handler)
 
 default_layout = {"widget": bokeh_garden.tabs.Tabs,
                   "Chose file": {"widget": bokeh.models.layouts.Column,
                                  "children": [{"widget": examples.grib_upload.GribUpload},
                                               {"widget": examples.grib_info_download.GribInfoDownload}]},
-                  "Plot": {"widget": examples.grib_plot.GribPlot}}
+                  "Plot": {"widget": examples.grib_plot.GribPlot},
+                  "Log": {"widget": examples.grib_log.GribLog}}
```
Comment (redhog) [inline examples/weather_data_explorer.py]: Change to bokeh_garden.LoggingBG

**Review by redhog — CHANGES_REQUESTED** [review]

File: examples/grib_plot.py (hunk 5 above)
Comment (glinares-em) [reply examples/grib_plot.py:18]: done

**Review by glinares-em — COMMENTED** [review]
All changes done

File: examples/weather_data_explorer.py [hunk 9]
```diff
@@ -3,16 +3,25 @@
 import examples.grib_plot
 import examples.grib_upload
 import examples.grib_info_download
+import logging
+import examples.grib_log
```
Comment (redhog) [inline examples/weather_data_explorer.py]: examples.grib_log isn't used any more and doesn't even exist.

However bokeh_garden.logging_bg needs to be included instead.

**Review by redhog — APPROVED** [review]

**Pull request merged**